docker/Dockerfile.spark

```diff
@@ -0,0 +1,51 @@
+# builder step used to download and configure spark environment
+FROM openjdk:11.0.11-jre-slim-buster as builder
```

Comment:
Would it be better to build the image on the basis of the official `apache/spark` image? If this docker file is referenced from some existing practice, could you please add its link to the description of this PR?

Reply:
I referred to this article[1] to write the dockerfile which is more controllable than the official image and works well.

[1] https://cloudinfrastructureservices.co.uk/create-apache-spark-docker-container-using-docker-compose/

Reply:
@jiangxin369 Using apache/spark would be simpler than compiling Spark from source. Also note that our Flink dockerfile did not compile Flink from source.

Can you explain why we should compile Spark from source?

Reply:
I searched for the existing solution of deploying spark cluster with docker-compose and most of them are compiling 
 from the source, so I just use it for saving time. Indeed it would be cleaner to use apache/spark after testing, so I changed it.